### PR TITLE
project-survey-vite-Amanda-Huadan

### PR DESCRIPTION
**Pull request by AmandaRam**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,3 +1,150 @@
...
+    const setPreparation = (event) => {
+        const { value } = event.target;
+        setForm({
+            ...form, // Keeping the existing form values
+            prepareMushroom: form.prepareMushroom.includes(value)
+                // If the value is already in prepareMushroom array, remove it. Otherwise, add it.
+                ? form.prepareMushroom.filter((wayOfPreparation) => wayOfPreparation !== value)
+                : [...form.prepareMushroom, value],
+        });
+    }
```
Comment (VAstrom) [inline src/App.jsx:69]: This block of code took some time to understand and is really advanced. Nice job there!

**Review by VAstrom — APPROVED** [review]
Good job with the project!
The layout with the mushroom picture is really nice! 
The code is really advanced and I'm not sure I understand all of it yet! I did notice a small "bug" when only choosing one checkbox option for the mushroom preparation. The summary at the end shows "or" even if only one option is chosen.

Apart from this, great work this week! :)

File: src/components/Card.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import styles from "./Card.module.css";
+
+const Card = ({ children }) => {
+    return <div className={styles.card}>{children}</div>;
+};
+
+export default Card;
```
Comment (Sheryll-Ann) [inline src/components/Card.jsx:7]: This code syntax is hard for me to grasp.  Is it that you are creating a common css style applied to all the children of your cards?

File: src/App.jsx [hunk 3]
```diff
@@ -1,3 +1,153 @@
...
+// Returns a nicely formatted string: Apple, Banana or Cherry
+const formatPreparation = (words) => {
+    if (words.length === 1)
+        return words[0].toLowerCase();
+
+    const listOfWords = [...words];
+    const lastWord = listOfWords.pop();
+
+    return `${listOfWords.join(", ").toLowerCase()} or ${lastWord.toLowerCase()}`;
+}
```
Comment (Sheryll-Ann) [inline src/App.jsx:21]: A very clever way to create this reusable function when needed, but may I ask when is this used in the code?

File: src/App.jsx [hunk 4]
```diff
@@ -1,3 +1,153 @@
...
+            yearsPickingMushrooms: "",
+        });
+
+        setStep("form")
+    };
+
+    // Sets a value in our form based on the name property of our target input.
+    const setFormValue = (event) => {
+        // Destructuring event.target to get the 'name' and 'value' properties.
+        const { name, value } = event.target;
```
Comment (Sheryll-Ann) [inline src/App.jsx:53]: This is advanced javascript code writing, and did not know that two functions could be combined in one single code line this way.

File: src/App.jsx [hunk 5]
```diff
@@ -1,3 +1,153 @@
...
+    const setPreparation = (event) => {
+        const { value } = event.target;
+        setForm({
+            ...form, // Keeping the existing form values
+            prepareMushroom: form.prepareMushroom.includes(value)
+                // If the value is already in prepareMushroom array, remove it. Otherwise, add it.
+                ? form.prepareMushroom.filter((wayOfPreparation) => wayOfPreparation !== value)
+                : [...form.prepareMushroom, value],
+        });
+    }
```
Comment (Sheryll-Ann) [inline src/App.jsx:72]: Reminds me of the GuessWho game but this time using a ternary operator, hard to grasp until the link with checkbox is made to understand all the clickable/checked combinations.

File: src/App.jsx [hunk 6]
```diff
@@ -1,3 +1,153 @@
...
+
+    // Submit button is disabled if the form values are not filled in.
+    const submitIsDisabled = form.name === "" || form.favoriteMushroom === "" || form.yearsPickingMushrooms === "" || form.prepareMushroom.length === 0
```
Comment (Sheryll-Ann) [inline src/App.jsx:75]: Neat code writing in one single function regrouping all form inputs to disable submit button.

File: src/App.jsx [hunk 7]
```diff
@@ -1,3 +1,153 @@
...
+                            value={form.favoriteMushroom}
+                            onChange={setFormValue}
+                            options={[
+                                "Chanterelle",
+                                "Porcini",
+                                "Black Trumpet",
+                                "Morel",
+                            ]}
+                        />
+                    </Card>
```
Comment (Sheryll-Ann) [inline src/App.jsx:121]: Amazing to see how the setFormValue update can be used both for text input, radio buttons and select tags and how the code writing is modified subsequently...

File: src/components/TextInput.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+            type="text"
+            name={name}
+            value={value}
+            onChange={onChange}
+            placeholder={placeholder}
+        />
+    );
+};
+
+export default TextInput;
```
Comment (Sheryll-Ann) [inline src/components/TextInput.jsx:23]: May I ask if there is any reason for using the previous version code writing of React in all your components in contrary to App.jsx using the newest version?  Maybe just an omission when combining the codes?

File: src/components/Container.jsx [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import styles from "./Container.module.css";
+
+const Container = ({ children }) => {
+    return <div className={styles.container}>{children}</div>;
+};
```
Comment (Sheryll-Ann) [inline src/components/Container.jsx:5]: same functionality as for the cards, thank you for sharing.

**Review by Sheryll-Ann — COMMENTED** [review]
Very advanced professional code writing.  Thank you for sharing it Amanda and Huadan.

File: src/components/TextInput.jsx (hunk 8 above)
Comment (AmandaRam) [reply src/components/TextInput.jsx:23]: I just forgot to make the App a default exported component. I prefer to use default export for components that only export a single component :) You can read more about it here: https://medium.com/@heshramsis/understanding-the-difference-between-export-default-and-export-with-named-exports-in-javascript-f0569c221a3 @Sheryll-Ann

File: src/components/Card.jsx (hunk 2 above)
Comment (AmandaRam) [reply src/components/Card.jsx:7]: Using CSS modules you can create local classes that are applied to only the elements inside of that component. With CSS modules you can define multiple instances of a CSS class and they will not overwrite each other. Say for instance if you have two cards that you want style differently but use the same CSS class name you can use modules. React will give them a unique class name in the generated HTML code @Sheryll-Ann

File: src/App.jsx (hunk 6 above)
Comment (EvelynDelCarmen) [inline src/App.jsx:75]: Ease to understand, nice!

File: src/components/Select.jsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+}) => {
+    return (
+        <select
+            id={name}
+            className={styles.select}
+            name={name}
+            value={value}
+            onChange={onChange}
+        >
+            <option value="" disabled>Please choose an option</option>
```
Comment (EvelynDelCarmen) [inline src/components/Select.jsx:17]: I actually understand this code :), the disable part!

**Review by EvelynDelCarmen — COMMENTED** [review]
Nice work! 💫 Overall nice structure and easy to understand! //Evelyn

**Pull request closed without merging**